format/spec.md

```diff
@@ -1761,6 +1764,10 @@ The reference Java implementation uses a type 4 uuid and XORs the 4 most signifi
 
 Java writes `-1` for "no current snapshot" with V1 and V2 tables and considers this equivalent to omitted or `null`. This has never been formalized in the spec, but for compatibility, other implementations can accept `-1` as `null`. Java will no longer write `-1` and will use `null` for "no current snapshot" for all tables with a version greater than or equal to V3.
 
+### Legacy naming for GZIP compressed Metadata JSON files
+
+Some implementations have written GZIP compressed metadata JSON files with the suffix `metadata.json.gz`. The reference Java implementation will interpret files with this naming convention as GZIP files.
```

Comment:
reference 
https://github.com/apache/iceberg/blob/74051b44f933bf3f33bb9ab9b04ed987e16b40dc/core/src/main/java/org/apache/iceberg/TableMetadataParser.java#L68-L74

Reply:
I think we should also mention that `.metadata.json.gz` is supported for backwards compatibility

Reply:
I thought that was implied here?  Could you suggest some language here?

Reply:
it does imply that! I think its better to be explicit

nit:
```suggestion
Some implementations have written GZIP compressed metadata JSON files with the suffix `metadata.json.gz`. The reference Java implementation will interpret files with this naming convention as GZIP files for backwards compatibility.
```